lisa/tests/staging/sched_android.py

```diff
@@ -0,0 +1,290 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (C) 2019, Arm Limited and contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import os
+import os.path
+import abc
+
+from lisa.wlgen.rta import RTA, Periodic
+from lisa.tests.base import ResultBundle, CannotCreateError, RTATestBundle
+from lisa.trace import Trace, FtraceCollector, FtraceConf, requires_events
+from lisa.target import Target
+from lisa.utils import ArtifactPath
+
+class SchedTuneTest(RTATestBundle, abc.ABC):
+    """
+    Abstract class for SchedTune testing
+
+    :param boost: The SchedTune `boost` level to apply the rtapp's cgroup
+    :type boost: int
+
+    :param prefer_idle: The SchedTune `prefer-idle` flag to apply to the rtapp's
+        cgroup
+    :type prefer_idle: bool
+
+    This class provides :meth:`get_cgroup` to create in the SchedTune
+    controller a `lisa_test` cgroup configured with the `boost` and
+    `prefer_idle` attributes.
+    """
+
+    boost = 0
+    prefer_idle = False
+
+    @classmethod
+    def get_cgroup_configuration(cls, plat_info):
+        cg = {
+                'controller': 'schedtune',
+                'name': 'lisa_test',
+                'values': {
+                    'boost': cls.boost,
+                    'prefer_idle': int(cls.prefer_idle)
+                }
+             }
+
+        return cg
+
```

Comment:
You have to define and annotate `from_target()` here for exekall to pick it up (which I think is documented nowhere... On the list it goes!). I got exekall to pick them up with this:

```diff
diff --git a/lisa/tests/staging/sched_android.py b/lisa/tests/staging/sched_android.py
index 8d8678f2..ebb3055d 100644
--- a/lisa/tests/staging/sched_android.py
+++ b/lisa/tests/staging/sched_android.py
@@ -57,6 +57,11 @@ class SchedTuneTest(RTATestBundle, abc.ABC):
 
         return cg
 
+    @classmethod
+    def from_target(cls, target:Target, res_dir:ArtifactPath=None,
+                    ftrace_coll:FtraceCollector=None) -> 'SchedTuneTest':
+        return super().from_target(target, res_dir, ftrace_coll=ftrace_coll)
+
 class SchedTuneFreqBoost(SchedTuneTest, abc.ABC):
     """
     Abstract class for testing SchedTune's frequency biasing feature
```
(when no docstring is specified and we're overriding a method, we use the docstring of the super() method, so you don't have to write one).

Reply:
We could consider having that "basic" annotated version directly in RTATestBundle, but so far we rejected that on the ground that if you need additional parameters, it will fail strangely at runtime (missing parameter for `_from_target()` exception). I don't feel strongly against or for any of these options.